src/auslib/db.py

```diff
@@ -168,7 +168,8 @@ def verify_signoffs(potential_required_signoffs, signoffs):
     for signoff in signoffs:
         signoffs_given[signoff["role"]] += 1
     for rs in potential_required_signoffs:
-        required_signoffs[rs["role"]] = max(required_signoffs.get(rs["role"], 0), rs["signoffs_required"])
+        signoff_verifier = "role" if "role" in rs else "permission"
```

Comment:
Unless this function needs to handle verifying permission _and_ non-permission signoffs at the same time, I think it would be better to have this passed as an argument.

It may also be worth considering moving this into a method on the `RequiredSignoffsTable`, and having `ProductRequiredSignoffsTable` and `PermissionsRequiredSignoffsTable` wrap it to set `signoff_verifier` correctly, as we already have those two classes that specialize signoffs along the exact lines we're doing here.

Reply:
Not sure I understand what you mean by wrap it.

 Does this mean setting the signoff_verifier in the `ProductRequiredSignoffsTable` and `PermissionsRequiredSignoffsTable` and then passing the signoff_verifier to `verify_signoffs`?